src/node_sqlite.cc

```diff
@@ -1804,15 +1832,29 @@ void StatementSync::Iterate(const FunctionCallbackInfo<Value>& args) {
 
   auto is_finished_pd =
       v8::PropertyDescriptor(v8::Boolean::New(isolate, false), true);
-  stmt_pd.set_enumerable(false);
-  stmt_pd.set_configurable(false);
+  is_finished_pd.set_enumerable(false);
+  is_finished_pd.set_configurable(false);
   if (iterable_iterator
           ->DefineProperty(context, env->isfinished_string(), is_finished_pd)
           .IsNothing()) {
     // An error will have been scheduled.
     return;
   }
 
+  // Add the return_arrays flag to the iterator
+  auto return_arrays_pd = v8::PropertyDescriptor(
```

Comment:
I don't think we should expose state to JavaScript land like this. In general, the iterator implementation needs to be refactored, but IMO this would not be a step in the right direction.

Reply:
I wouldn't block on this because, as you mentioned, this is copying the existing way of doing things.